### PR TITLE
refactor: remove repetitive folds in `Z3Context`

### PR DESCRIPTION
**Pull request by konnov**
I have noticed a few repetitive occurrences of summing up longs. Refactored that to a single call.

### CODE REVIEW

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala [hunk 1]
```diff
@@ -767,26 +767,17 @@ class Z3SolverContext(val config: SolverConfig) extends SolverContext with LazyL
       case OperEx(ApalacheInternalOper.distinct, args @ _*) =>
         val (es, ns) = (args.map(toExpr)).unzip
         val distinct = z3context.mkDistinct(es: _*)
-        (distinct.asInstanceOf[ExprSort],
-            ns.foldLeft(1L) {
-              _ + _
-            })
+        (distinct.asInstanceOf[ExprSort], sumExprCount(ns))
```
Comment (thpani) [inline tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala:770]:
```suggestion
        (distinct.asInstanceOf[ExprSort], ns.sum + 1)
```

**Review by thpani — COMMENTED** [review]
Would it be possible / easier to use [`Seq.sum`](https://www.scala-lang.org/api/2.12.2/scala/collection/immutable/Seq.html#sum:A)?

File: tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala (hunk 1 above)
Comment (konnov) [reply tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/smt/Z3SolverContext.scala:770]: Oh, I have missed that `Seq.sum` existed. That's definitely a better alternative!

**Comment by konnov** [comment]
> Would it be possible / easier to use [`Seq.sum`](https://www.scala-lang.org/api/2.12.2/scala/collection/immutable/Seq.html#sum:A)?

Fixed that. Please review

**Review by thpani — APPROVED** [review]

**Pull request merged**